### PR TITLE
fix: Correct indentation of all the XML/JSON/yaml and etc. files

### PR DESCRIPTION
**Pull request by joeylimd**
Fixes: #189

### CODE REVIEW

File: runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public void testGenerateListMappingBasic() throws Exception {
         AtlasMappingService atlasMappingService = new AtlasMappingService(
                 Arrays.asList("io.atlasmap.v2", "io.atlasmap.java.v2", "io.atlasmap.xml.v2"));
         atlasMappingService.saveMappingAsFile(a,
-                new File("src/test/resources/javaToJava/atlasmapping-complex-list-autodetect-base.xml"));
+                new File("target/atlasmapping-complex-list-autodetect-base.xml"));
```
Comment (rohanmars) [inline runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java]: why have these paths changed?

Comment (joeylimd) [reply runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java]: 1. In the java-generated xml files, the indentation is 4 spaces, which does not align with the new standard - 2 spaces
2. Those are existing xml files. Unit tests should not modify pre-defined testing files under /src/test folder. All new output files should be saved under target folder.
3. If the path is not changed, the new generated xml files with 4-space indentation will break code format validation. That's why the path got changed.

Comment (droneware) [reply runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java]: The tests should not be overriding source files in the first place. The paths should keep the original output structure but within the target directory. I.e. target/reference-mappings/javaToJava/......xml

Comment (rohanmars) [reply runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java]: The tests should not be overriding source files in the first place. The paths should keep the original output structure but within the target directory. I.e. target/reference-mappings/javaToJava/......xml

Comment (joeylimd) [reply runtime/itests/reference-mappings/src/test/java/io/atlasmap/reference/java_to_java/JavaJavaComplexListTest.java]: Done. Generating output files within target sub directory. Please review.

**Comment by rohanmars** [comment]
@joeylimd, can you rebase this branch and resolve any conflicts.

**Comment by rohanmars** [comment]
LGTM.  :+1: 

* only a random selection of reference mappings was preformed due to the number of files.

**Comment by rohanmars** [comment]
@igarashitm would you want the xml-maven-plugin to error out if there are errors in the xml files?  Or just a warning and move on?

**Comment by igarashitm** [comment]
Could it easily exclude the intentionally invalid XML files for test purposes?

**Comment by igarashitm** [comment]
It might be better exclude everything in src/test/resources

**Comment by joeylimd** [comment]
1. to exclude a specific new invalid XML files, a new entry needs to be added in runtime POM.xml every time, e.g. <exclude>pathToFile/foo.ff</exclude>. 
2. the issue is caused by file atlasmap\runtime\service\src\test\resources\mappings\bar.xml and foo.xml although they are excluded. xml-mven-plugin still reads these 2 files, then throws Exception Premature end of file, which breaks build as per config failOnFormatViolation is true. from this thread, https://stackoverflow.com/questions/10022796/why-am-i-getting-this-error-premature-end-of-file, sounds it's a bug of this plugin.
3. If failOnFormatViolation is false, it generate warning messages without breaking build.

**Comment by igarashitm** [comment]
That sounds like something we don't want to enable at all. Even warnings are annoying.

**Comment by rohanmars** [comment]
Maybe put the plugin declaration in new maven profile in the pom.xml. That way it will only be enabled when specifically activated.

**Comment by igarashitm** [comment]
I'm OK with the profile. But can you also completely exclude any src/test/resources? I think we don't care if test resources are well formatted, and adding each intentional error files into POM is really annoying.

**Comment by joeylimd** [comment]
1. in profile, excluded files in src/test/resources, and adding each intentional error files into POM is not needed.
2. The roadblock with xml-maven-plugin is that XML can't be excluded, but all other type of files can be excluded. With profile, if needed, it can be enabled with **mvn -Pxml-format**.
Is it fine to move on? Thanks again.

**Comment by igarashitm** [comment]
Yes that sounds reasonable. Also please don't forget to rebase against current upstream master :-)

**Comment by igarashitm** [comment]
Ah hold on please, I want to rollback the version number to `1.33-SNAPSHOT` in the POM.

**Comment by joeylimd** [comment]
@igarashitm sure. notice there are many conflicts in POM file. if rollback to 1.33-snapshot, that will save a lot of time. Sweet... when done, please post it. Thanks.

**Comment by igarashitm** [comment]
Ha, look at this! conflicts were just that version number :-)

**Comment by joeylimd** [comment]
@igarashitm Thanks a bunch again. It's ready to get merged.
1. Totally 484 files got formatted. No files are deleted this time.
2. To check xml format with profile xml-format, run **mvn -Pxml-format clean install** 
3. To not check xml format, normally run mvn clean install
4. To generate warning messages only, instead of format-check failure, please set **failOnFormatViolation** as **false** in **atlas-runtime POM.xml**

**Pull request merged**